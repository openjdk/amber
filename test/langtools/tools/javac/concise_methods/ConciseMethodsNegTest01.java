/*
 * @test /nodynamiccopyright/
 * @summary smoke test for the concise method feature
 * @compile/fail/ref=ConciseMethodsNegTest01.out -XDrawDiagnostics ConciseMethodsNegTest01.java
 */

class ConciseMethodsNegTest01 implements Runnable {
    int length(String s) = s::length;

    abstract int length2(String s) = String::length;

    native int length3(String s) = String::length;

    static ConciseMethodsNegTest01 make() {
        return new ConciseMethodsNegTest01();
    }

    ConciseMethodsNegTest01() = ConciseMethodsNegTest01::make;

    private static void all() {}

    public static void foo() = ConciseMethodsNegTest01.all();

    public void run() -> null;

    Runnable f(Runnable r) -> r;

    void m() = f(this)::run;
}
