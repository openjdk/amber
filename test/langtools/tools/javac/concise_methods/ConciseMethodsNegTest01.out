ConciseMethodsNegTest01.java:7:1: compiler.err.does.not.override.abstract: ConciseMethodsNegTest01, length2(java.lang.String), ConciseMethodsNegTest01
ConciseMethodsNegTest01.java:8:28: compiler.err.cant.resolve.location: kindname.variable, s, , , (compiler.misc.location: kindname.class, ConciseMethodsNegTest01, null)
ConciseMethodsNegTest01.java:10:18: compiler.err.abstract.meth.cant.have.body
ConciseMethodsNegTest01.java:12:16: compiler.err.native.meth.cant.have.body
ConciseMethodsNegTest01.java:18:5: compiler.err.constructors.cant.have.concise.body
ConciseMethodsNegTest01.java:22:59: compiler.err.only.method.references.allowed
ConciseMethodsNegTest01.java:28:17: compiler.err.expression.must.be.type.or.constant.or.final.field
7 errors
